tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java

```diff
@@ -0,0 +1,377 @@
+    // resource path for clearing state
+    public static final String RESET_PATH = "/reset";
+
+    // headers for tests to indicate how the participant should respond to callbacks
+    // from the LRA implementation (complete, compensate, forget and status)
+    public static final String LRA_TCK_FAULT_TYPE_HEADER = "Lra-Tck-Fault-Type";
+    public static final String LRA_TCK_FAULT_CODE_HEADER = "Lra-Tck-Fault-Status";
+
+    // a header for tests to indicate which LRA stats are being queried/reset
+    public static final String LRA_TCK_HTTP_CONTEXT_HEADER = "Lra-Tck-Context";
+
+    private static final String REQUIRES_NEW_LRA_PATH = "/requires-new-lra";
+
+    /**
+     * A class to hold all of the metrics gathered in the context of a single LRA.
+     * We need stats per LRA since a misbehaving test may interfere with subsequent tests
+     */
+    private static class LRAMetric {
+        String lraId;
+        Map<String, AtomicInteger> metrics = Stream.of(
+                new AbstractMap.SimpleEntry<>(Complete.class.getName(), new AtomicInteger(0)),
+                new AbstractMap.SimpleEntry<>(Compensate.class.getName(), new AtomicInteger(0)),
+                new AbstractMap.SimpleEntry<>(Status.class.getName(), new AtomicInteger(0)),
+                new AbstractMap.SimpleEntry<>(Forget.class.getName(), new AtomicInteger(0)))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        LRAMetric(String lraId) {
+            this.lraId = lraId;
+        }
+
+        void increment(String metric) {
+            if (metrics.containsKey(metric)) {
+                metrics.get(metric).incrementAndGet();
+            }
+        }
+
+        int get(String metric) {
+            if (metrics.containsKey(metric)) {
+                return metrics.get(metric).get();
+            }
+
+            return -1;
+        }
+
+        void clear() {
+            metrics.forEach((k, v) -> v.set(0));
+        }
+    }
+
+    // maintain metrics on a per LRA basis
+    private Map<String, LRAMetric> metrics = new HashMap<>();
+
+    private void incrementMetric(String name, String lraId) {
+        metrics.putIfAbsent(lraId, new LRAMetric(lraId));
+        metrics.get(lraId).increment(name);
+    }
+
+    /**
+     * An enum which controls the behaviour of participant when the
+     * LRA spec implementation invokes the compensation, completion,
+     * forget and status callbacks. It is used for testing the
+     * implementation is spec compliant.
+     */
+    public enum EndPhase {
+        ACCEPTED, // a participant reports that the complete/compensate is in progress
+        FAILED, // a particant reports that is is unable to complete/compensate
+        SUCCESS // clear the any injected behaviour
+    }
+
+    private EndPhase endPhase = EndPhase.SUCCESS;
+    // control which status to return from participant callbacks
+    private Response.Status endPhaseStatus = Response.Status.OK;
```

Comment:
There is only ever one LRA in play for each test. The reason I maintain a collection of LRAids is:

>    /**
>    * A class to hold all of the metrics gathered in the context of a single LRA.
>     * We need stats per LRA since a misbehaving test may interfere with subsequent tests
>     */

Reply:
To further clarify why this could be a problem: a misbehaving test may leave an LRA in need of recovery which means that the compensate/complete call will continue to be called when subsequent tests run - ie it is not possible to fully tear down a failing test.

Reply:
I have just pushed an update that includes that extra explanatory text.